### PR TITLE
Upgrade Pulfalight Ruby to 3.1

### LINKED ISSUE #3344: Update pulfalight to ruby 3.1
Ruby 2.x is EOL soon. Use Ruby 3.1

### PR DESCRIPTION
**Pull request by eliotjordan**
Closes #3344

### CODE REVIEW

File: group_vars/pulfalight/production.yml [hunk 1]
```diff
@@ -17,8 +17,8 @@ solr_mirror: "https://archive.apache.org/dist"
 passenger_server_name: "findingaids.princeton.edu"
 passenger_app_root: "/opt/pulfalight/current/public"
 passenger_app_env: "production"
-passenger_ruby: "/usr/bin/ruby2.7"
-ruby_version_override: "ruby2.7"
+passenger_ruby: "/usr/bin/ruby3.1"
```
Comment (kayiwa) [inline group_vars/pulfalight/production.yml:20]: you know about the `ruby_s` role?

Comment (eliotjordan) [reply group_vars/pulfalight/production.yml:20]: I'm having some difficulty with this and set it down at the end of the day. I did try the `ruby_s` role, but then the playbook also ran the `ruby` role afterwards and failed.

Comment (kayiwa) [reply group_vars/pulfalight/production.yml:20]: always happy to pair. If @acozine and I ran the world we'd hoped moving to ruby 3 would sunset the ruby role altogether.

Comment (eliotjordan) [reply group_vars/pulfalight/production.yml:20]: Thanks. I think I've figured it out. The passenger role already depends on the `ruby` or `ruby_s` role so we don't need to explicitly call it in the pulfalight playbook. Updated PR coming shortly...

**Review by hackartisan — APPROVED** [review]

**Pull request merged**